source/solvers/navier_stokes_base.cc

```diff
@@ -1668,8 +1669,36 @@ NavierStokesBase<dim, VectorType, DofsType>::write_output_results(
   VorticityPostprocessor<dim> vorticity;
   data_out.add_data_vector(solution, vorticity);
 
-  QCriterionPostprocessor<dim> qcriterion;
-  data_out.add_data_vector(solution, qcriterion);
+  QCriterionPostprocessor<dim>                      qcriterion;
+  QcriterionPostProcessorSmoothing<dim, VectorType> qcriterion_smoothing(
+    this->triangulation,
+    this->simulation_parameters,
+    number_quadrature_points,
+    this->mpi_communicator);
+
+  if (!this->simulation_parameters.post_processing.smoothing)
+    {
+      data_out.add_data_vector(solution, qcriterion);
+    }
+  else
```

Comment:
I think it would be easier to read it the other way. If smoothing, else. But that's your final call. :)

Reply:
Absolutely. I changed it.